spyder/utils/introspection/docstrings.py

```diff
@@ -0,0 +1,285 @@
+    ['threading.Thread']
+    >>> bool(_search_param_in_docstr('no document', 'param'))
+    False
+    >>> _search_param_in_docstr(':param int param: some description', 'param')
+    ['int']
+    """
+    # look at #40 to see definitions of those params
+
+    # Check for Sphinx/Epydoc params
+    patterns = [re.compile(p % re.escape(param_str))
+                for p in DOCSTRING_PARAM_PATTERNS]
+
+    found = None
+    for pattern in patterns:
+        match = pattern.search(docstr)
+        if match:
+            found = [_strip_rst_role(match.group(1))]
+            break
+    if found is not None:
+        return found
+
+    # Check for numpy style params
+    found = _search_param_in_numpydocstr(docstr, param_str)
+    if found is not None:
+        return found
+
+    return []
+
+
+def _strip_rst_role(type_str):
+    """
+    Strip off the part looks like a ReST role in `type_str`.
+
+    >>> _strip_rst_role(':class:`ClassName`')  # strip off :class:
+    'ClassName'
+    >>> _strip_rst_role(':py:obj:`module.Object`')  # works with domain
+    'module.Object'
+    >>> _strip_rst_role('ClassName')  # do nothing when not ReST role
+    'ClassName'
+
+    See also:
+    http://sphinx-doc.org/domains.html#cross-referencing-python-objects
+    """
+    match = REST_ROLE_PATTERN.match(type_str)
+    if match:
+        return match.group(1)
+    else:
+        return type_str
+
+
+def _evaluate_for_statement_string(evaluator, string, module):
+    if string is None:
+        return []
+
+    code = dedent("""
+    def pseudo_docstring_stuff():
+        # Create a pseudo function for docstring statements.
+    %s
+    """)
+
+    for element in re.findall('((?:\w+\.)*\w+)\.', string):
+        # Try to import module part in dotted name.
+        # (e.g., 'threading' in 'threading.Thread').
+        string = 'import %s\n' % element + string
+
+    # Take the default grammar here, if we load the Python 2.7 grammar here, it
+    # will be impossible to use `...` (Ellipsis) as a token. Docstring types
+    # don't need to conform with the current grammar.
+    p = Parser(load_grammar(), code % indent_block(string))
+    try:
+        pseudo_cls = p.module.subscopes[0]
+        # First pick suite, then simple_stmt (-2 for DEDENT) and then the node,
+        # which is also not the last item, because there's a newline.
+        stmt = pseudo_cls.children[-1].children[-2].children[-2]
+    except (AttributeError, IndexError):
+        type_list = []
+    else:
+        # Use the module of the param.
+        # TODO this module is not the module of the param in case of a function
+        # call. In that case it's the module of the function call.
+        # stuffed with content from a function call.
+        pseudo_cls.parent = module
+        type_list = _execute_types_in_stmt(evaluator, stmt)
+    return type_list
+
+
+def _execute_types_in_stmt(evaluator, stmt):
+    """
+    Executing all types or general elements that we find in a statement. This
```

Comment:
First line should be a one line description then a blank line then the extra details